From 2bf799928e9002b991c22a7ba6f9024676b25584 Mon Sep 17 00:00:00 2001
From: Arian <arian.kulmer@web.de>
Date: Sat, 26 Apr 2025 17:58:26 +0200
Subject: [PATCH 03/10] Add Ultra HDR heic configurations to
 StreamConfigurationMap constructor

Commit "Add HEIC_ULTRAHDR image format"[1] added three new parameters to the constructor of StreamConfigurationMap.
Since HyperOS 2, xiaomi has added these methods but guarded them for sdk 36 and above.
Android 15 QPR2 (still sdk 35) added these methods already, so enforce the usage of the new constructor.

[1]: https://github.com/LineageOS/android_frameworks_base/commit/ba41b51131af923a797480a6e707d390d39852c2
---
 smali_classes4/mg/c.smali | 24 +-----------------------
 1 file changed, 1 insertion(+), 23 deletions(-)

diff --git a/smali_classes4/mg/c.smali b/smali_classes4/mg/c.smali
index e182c89c7..0a87f185a 100644
--- a/smali_classes4/mg/c.smali
+++ b/smali_classes4/mg/c.smali
@@ -553,31 +553,9 @@
 .end method
 
 .method public static e()Z
-    .locals 2
-
-    sget v0, Landroid/os/Build$VERSION;->SDK_INT:I
-
-    const/16 v1, 0x23
-
-    if-gt v0, v1, :cond_1
-
-    if-ne v0, v1, :cond_0
-
-    sget v0, Landroid/os/Build$VERSION;->PREVIEW_SDK_INT:I
-
-    if-lez v0, :cond_0
-
-    goto :goto_0
-
-    :cond_0
-    const/4 v0, 0x0
+    .locals 1
 
-    goto :goto_1
-
-    :cond_1
-    :goto_0
     const/4 v0, 0x1
 
-    :goto_1
     return v0
 .end method
-- 
2.47.2

